### PR TITLE
feat(scales): plan semantic temporal guides

### PR DESCRIPTION
**Pull request by ljodea**
## Summary

- add portable calendar interval controls (`dateBreaks`, `dateMinorBreaks`, `dateLabels`, `locale`, and `weekStart`) across schema, runtime validation, helpers, declarations, and docs
- add immutable semantic `AxisGuidePlan`s with measured two-pass temporal planning, same-or-coarser hysteresis, calendar-aligned major/minor ticks, contextual visible labels, and complete standalone labels
- thread plans through single and faceted layouts, `RenderModel.guidePlans`, `ScaleDecision.guidePlanIds`, scene grids/ticks, SVG output, and Svelte axis chrome
- preserve explicit authored breaks and labels without silent thinning, truncation, or ellipsis; add stable overlap, margin, progression, limit, and out-of-domain diagnostics
- add locale/timezone/DST handling, inferred-axis formatters, SSR/core equality, first-child `<title>` accessibility, and privacy-safe Playground guide reports
- ship docs, ADR 0014, eval case 48, packed-consumer coverage, four temporal benchmark gates, and responsive browser/ggplot2 evidence

## Contract

- PortableSpec remains canonical JSON with closed interval and label grammars.
- Temporal parsing remains pre-stat; guide planning follows scale training; coordinate transforms remain post-stat.
- Automatic Pass B can retain Pass A or move coarser, never finer or to a third pass.
- Explicit labels remain byte-identical even when overlap or margin diagnostics fire.
- Deterministic defaults remain `en-US`, UTC, and Monday week starts.

## Test plan

- [x] pre-push repository gates pass: package builds, docs and package `svelte-check`, type-aware oxlint, examples TypeScript, knip, actionlint, zizmor, and the spec/core/benchmark/script/eval harness
- [x] spec/core/benchmark/script/eval suite: 1,220 tests passed during final implementation verification
- [x] Svelte browser suite: 210 files, 2,847 tests passed
- [x] Svelte SSR suite: 15 tests passed
- [x] Playground/temporal Playwright: 8 tests passed
- [x] deterministic evals: 48/48 passed
- [x] packed npm consumer compatibility passed with Svelte 5.33.1
- [x] generated route/search projections pass 5/5 and both freshness checks pass
- [x] all four new temporal benchmark gates pass; isolated CI `run-bench` remains authoritative for whole-suite absolute budgets

## Browser and performance evidence

Evidence: [`artifacts/scale-equivalence/pr2-temporal-guides/`](artifacts/scale-equivalence/pr2-temporal-guides/README.md)

- 320, 640, and 1200 px light/dark captures have four collision-free 50-year labels, positive measured gaps, zero horizontal overflow, and complete first-child `<title>` labels.
- The untouched 1835–2025 raw-year example infers temporal meaning without preprocessing or an explicit scale.
- The Playground report includes decision/guide metadata while excluding rows, field names, domains, labels, and source epochs.
- Candidate selection, resize churn, DST-heavy planning, and 100 free-facet layout are within committed budgets.
- No runtime dependency or lockfile change.

## Independent pre-PR reviews

Initial Grok and Claude reviews blocked on DST anchors, non-Latin formatting, locale propagation, inferred-axis formatting, facet hysteresis, schema/runtime parity, evidence, and related edge cases. Every finding is fixed or rebutted with tests and evidence in the [resolution ledger](artifacts/scale-equivalence/pr2-temporal-guides/pre-pr-review-resolution.md).

- Grok final full review: `VERDICT: PASS`
- Claude final full review: `VERDICT: PASS`
- Grok supplemental generated-doc review at `0519ab0`: no findings, `VERDICT: PASS`
- Claude supplemental generated-doc review at `0519ab0`: no findings, `VERDICT: PASS`

No unresolved P0/P1 findings.

## Changeset

Minor changesets for `@ggsvelte/spec`, `@ggsvelte/core`, and `@ggsvelte/svelte`.

Migration: none — additive.

## Visual baselines

No files under `tests/visual/__screenshots__/` are included. If baseline publication is needed, it will follow the source-first `vr-approve` flow after merge.

### CODE REVIEW

File: packages/core/src/layout/format.ts [hunk 1]
```diff
@@ -108,6 +114,291 @@ export function formatTime(ms: number, pattern: string): string {
...
+  return formatter;
+}
+
+function partValue(parts: readonly Intl.DateTimeFormatPart[], type: Intl.DateTimeFormatPartTypes) {
+  return parts.find((part) => part.type === type)?.value ?? "";
+}
+
+function displayParts(ms: number, options: TemporalLabelFormatOptions): TemporalDisplayParts {
+  const locale = options.locale ?? "en-US";
+  const timezone = options.kind === "date" ? "UTC" : (options.timezone ?? "UTC");
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/core/src/layout/format.ts]: Fixed in `65ab6ac`. Temporal label formatting canonicalizes accepted `Z` and `Etc/UTC` aliases to `UTC` before formatter caching or `Intl.DateTimeFormat` construction. The regression uses `en-GB` to bypass the UTC fast path and proves both aliases render `2024-01-02 03:04 UTC` instead of throwing.

File: packages/spec/src/temporal-guides.ts [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+        ...parts,
+        hour: 0,
+        minute: 0,
+        second: 0,
+        millisecond: 0,
+      }).subtract({ days: daysBack });
+      break;
+    }
+    case "month": {
+      const month = Math.floor((parts.month - 1) / interval.step) * interval.step + 1;
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/spec/src/temporal-guides.ts]: Fixed in `65ab6ac`. Zoned month/quarter alignment now floors an absolute month index (`year * 12 + month - 1`) and reconstructs the civil boundary, matching the existing UTC anchor. Regressions compare UTC and America/New_York year/month sequences for both `18 months` and `5 quarters`.

File: packages/spec/src/validate-data-checks.ts [hunk 3]
```diff
@@ -75,24 +80,70 @@ export function dataChecks(
         config?.temporalKind !== undefined ||
         config?.timezone !== undefined ||
         config?.disambiguation !== undefined ||
-        config?.parseFailure !== undefined)
+        config?.parseFailure !== undefined ||
+        config?.dateBreaks !== undefined ||
+        config?.dateMinorBreaks !== undefined ||
+        config?.dateLabels !== undefined ||
+        config?.locale !== undefined ||
+        config?.weekStart !== undefined)
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/spec/src/validate-data-checks.ts:88]: Fixed in `65ab6ac`. Tier-2 compatibility now reuses the complete `scaleRequestsTime(axis)` predicate, so `dateBreaks`, `dateMinorBreaks`, `dateLabels`, `locale`, and `weekStart` all request a time scale during field/type validation. Added regressions that map each guide-only option to nominal data and require `scale-type-mismatch`; the focused spec/core suite passes 33/33.

File: packages/spec/src/temporal-guides.ts [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+
+export function temporalLocaleConfigurationError(locale: string): string | null {
+  if (locale.length === 0 || locale.length > 128) {
+    return "locale must contain 1 through 128 characters";
+  }
+  try {
+    const canonical = Intl.getCanonicalLocales(locale);
+    if (canonical.length !== 1) return `invalid or unsupported locale ${JSON.stringify(locale)}`;
+    new Intl.DateTimeFormat(canonical[0], { timeZone: "UTC", year: "numeric" }).format(0);
+    return null;
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/spec/src/temporal-guides.ts:158]: Fixed in `65ab6ac`. Locale validation now requires `Intl.DateTimeFormat.supportedLocalesOf(..., { localeMatcher: "lookup" })` to recognize the canonical tag before constructing the formatter. `zz-ZZ` is rejected with the stable unsupported-locale diagnostic while supported locales such as `en-US` remain accepted.

File: packages/spec/src/schema.ts [hunk 5]
```diff
@@ -1175,6 +1177,25 @@ const SpecDeclarations = {
           },
         ),
       ),
+      dateBreaks: Type.Optional(TemporalIntervalSpecSchema),
+      dateMinorBreaks: Type.Optional(TemporalIntervalSpecSchema),
+      dateLabels: Type.Optional(
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/spec/src/schema.ts]: Fixed in `65ab6ac`. `DiscretePositionScaleOptions` now omits every temporal guide field. `scaleXDiscrete`/`scaleYDiscrete` strip temporal fields defensively for untyped JS callers, and band normalization strips the same fields while preserving ordinary options such as `reverse`. Added compile-time and runtime/normalize regressions; `check:type-contracts` passes.

**Comment by ljodea** [comment]
@codex review

File: packages/core/src/layout/format.ts [hunk 6]
```diff
@@ -108,6 +114,293 @@ export function formatTime(ms: number, pattern: string): string {
...
+  label: string;
+  fullLabel: string;
+}
+
+export function formatTemporalTickSequence(
+  values: readonly number[],
+  options: TemporalLabelFormatOptions & { interval: TemporalInterval; pattern?: string },
+): TemporalTickLabel[] {
+  const full = compileTemporalLabelFormat(
+    options.kind === "date" ? "%Y-%m-%d" : "%Y-%m-%d %H:%M:%S %Z",
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/core/src/layout/format.ts]: Fixed in 34d9241. Standalone datetime labels now include %L when the selected interval is millisecond-level or any tick retains a sub-second remainder. Tests prove distinct millisecond fullLabel values, retained precision under a coarser interval, and unchanged whole-second labels.

File: packages/core/src/pipeline/panel-layout-chrome-display.ts [hunk 7]
```diff
@@ -58,6 +58,38 @@ export function resolvePanelLayoutDisplay(input: PanelLayoutDisplayInput): Panel
   };
   const { hBreaks, vBreaks } = flipDisplayBreaks(flip, convertedBreaks("x"), convertedBreaks("y"));
   const { freeH, freeV } = flipDisplayFreeFlags(flip, freeX, freeY);
+  const displayScales = makeDisplayScalesFn(flip, panelScales);
+  const displayTemporal = (panelIndex: number) => {
+    const xKind = input.xTemporalKind ?? scalesConfig.x?.temporalKind ?? null;
+    const yKind = input.yTemporalKind ?? scalesConfig.y?.temporalKind ?? null;
+    const x =
+      xScale.type === "time" && xKind !== null
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/core/src/pipeline/panel-layout-chrome-display.ts:66]: Fixed in 34d9241. Temporal kind resolution now uses authored kind, mapped decisions, strict scalar annotation evidence, then a deterministic datetime fallback only for a trained requested time scale. A rowless ymd rule regression asserts source=interval, interval=3 days, exact epochs, kind=date, and no fabricated x ScaleDecision.

**Comment by ljodea** [comment]
@codex review

**Review by ljodea — COMMENTED** [review]
Review follow-up fixes verified.

File: packages/core/src/layout/temporal-guide.ts [hunk 8]
```diff
@@ -0,0 +1,493 @@
...
+            ? -1
+            : left.interval.key > right.interval.key
+              ? 1
+              : 0),
+      );
+  const primary = evaluatePool(
+    AUTOMATIC_INTERVALS.filter((candidate) => {
+      const approx = intervalApproxMs(candidate);
+      return approx >= lowerApprox && approx <= upperApprox;
+    }),
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/core/src/layout/temporal-guide.ts]: Fixed in 413128d. Automatic date-kind planning now filters every candidate ladder (primary, coarser fallback, and Pass-B retention) to day/week/month/quarter/year. Datetime automatic planning and authored `dateBreaks: "6 hours"` still use sub-day intervals. The focused regression `keeps automatic date guides day-or-coarser without restricting authored intervals` asserts all three contracts.

File: packages/core/src/layout/temporal-guide.ts [hunk 9]
```diff
@@ -0,0 +1,493 @@
...
+    ticks,
+    ...evaluateTicks(ticks, input),
+    count: ticks.length,
+    approxMs: intervalApproxMs(intervalValue),
+  };
+}
+
+function inferExplicitInterval(values: readonly number[]): TemporalInterval {
+  if (values.length < 2) return interval("year", 1);
+  const gap = Math.abs(values[1]! - values[0]!);
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/core/src/layout/temporal-guide.ts]: Fixed in 413128d. Explicit formatting context now uses the smallest non-zero adjacent gap in a sorted, deduplicated copy of every surviving break. The guide still builds ticks from authored order/values and still exposes public `interval: null`. The regression requires `Jan 2024`, `Jan 2025`, and `Feb` for the reported fixture.

File: packages/spec/src/temporal-guides.ts [hunk 10]
```diff
@@ -0,0 +1,427 @@
...
+      break;
+    }
+    case "day":
+      plain = Temporal.PlainDateTime.from({
+        ...parts,
+        hour: 0,
+        minute: 0,
+        second: 0,
+        millisecond: 0,
+      });
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/spec/src/temporal-guides.ts]: Fixed in 413128d. Zoned day and week cursors now align an absolute Gregorian civil-day index to the same epoch/weekday phase as the UTC path before the minimum skip and `compatible` timezone projection. The Jan 2 example is already on the UTC epoch phase (so Jan 4 is correct there); the defect reproduces from Jan 3, where the old zoned path yielded Jan 5/7/9 and the fixed path matches UTC at Jan 4/6/8. Regressions cover `2 days`, `2 weeks`, explicit week starts, and the existing DST gap/fold matrix.

File: packages/core/src/pipeline/finalize-layout-pass.ts [hunk 11]
```diff
@@ -33,28 +34,82 @@ export function finalizePanelLayoutPass(input: {
...
+      const decision = parseTemporalColumn(
+        scalarValues,
+        conversion.parser,
+        conversion.options,
+      ).decision;
+      if (decision.kind !== null) return decision.kind;
+    }
+
+    const scale = axis === "x" ? xTraining.scale : yTraining.scale;
+    return scale.type === "time" && conversion.requestedTime ? "datetime" : null;
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/core/src/pipeline/finalize-layout-pass.ts:77]: Fixed in 413128d. Rowless kind resolution now strictly parses configured domain values, then explicit breaks when needed, with the resolved axis parser/options before the deterministic datetime fallback. The reported `ymd` domain now yields `temporalKind: "date"` and date-only full labels, while the regression confirms no x `ScaleDecision` is fabricated.

File: packages/spec/src/schema.ts [hunk 12]
```diff
@@ -1175,6 +1177,25 @@ const SpecDeclarations = {
...
+        Type.String({
+          minLength: 1,
+          maxLength: 128,
+          description:
+            "Strict temporal label format. Supported tokens: %Y %y %m %b %B %d %e %a %A %H %I %M %S %L %p %q %z %Z %%.",
+        }),
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/spec/src/schema.ts]: Fixed in 413128d. `TemporalLabelSpecSchema` now derives an anchored closed-token pattern beside `TEMPORAL_LABEL_TOKENS`, is reused by `PositionScaleSpec`, and is present in regenerated `schema/v0.json`. Tier-1 TypeBox and emitted AJV tests reject `%Q` and trailing `%` while accepting all documented tokens and literals. `runPipeline` still preserves the stable `invalid-temporal-labels` structured diagnostic through pre-schema string preflight.

**Comment by ljodea** [comment]
@codex review

All five P2 findings are fixed in `413128d`, with focused red/green regressions, both Grok and Claude final reviews passing, 1,263/1,263 repository tests green, and all review threads resolved.

File: packages/spec/src/temporal-guides.ts [hunk 13]
```diff
@@ -0,0 +1,427 @@
...
+    default:
+      return [];
+  }
+
+  // Generated civil boundaries are synthetic, not parsed source values. Temporal's
+  // compatible policy keeps them monotonic through gaps/folds; source parsing has
+  // already applied the author's configured disambiguation contract.
+  let tick = plain.toZonedDateTime(timezone, { disambiguation: "compatible" });
+  if (tick.epochMilliseconds < min) {
+    tick = addTemporalInterval(tick, interval);
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/spec/src/temporal-guides.ts]: ### ❌ Invalid / not actionable: already fixed on main

Why: Merged head of #348 advances the civil cursor *before* each projection so a single gap adjustment cannot shift the multi-hour phase. Regression in `packages/spec/tests/temporal-interval.test.ts` pins even-hour ticks after spring-forward:

`["04:00", "06:00", "08:00"]` for `America/New_York`, min `2024-03-10T03:30:00-04:00`, `dateBreaks: "2 hours"`.

No further code change.

File: packages/spec/src/normalize.ts [hunk 14]
```diff
@@ -203,17 +203,32 @@ function normalizePositionScale(scale: PositionScaleSpec): PositionScaleSpec {
...
     return band;
   }
+  const hasTemporalGuideOption =
+    scale.dateBreaks !== undefined ||
+    scale.dateMinorBreaks !== undefined ||
+    scale.dateLabels !== undefined ||
+    scale.locale !== undefined ||
+    scale.weekStart !== undefined;
+  if ((scale.type === "linear" || scale.type === "log") && hasTemporalGuideOption) {
+    return { ...scale };
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/spec/src/normalize.ts:222]: ### ❌ Invalid / not actionable: already rejected on main

Why: Tier-2 validation and the pipeline both reject temporal guide options on explicit linear/log/band scales with `scale-type-mismatch` (`packages/spec/src/validate-data-checks.ts`). Covered by `rejects temporal guides on explicitly non-time runtime scales` in `pipeline-temporal.test.ts`.

Evidence: `runPipeline` with `{ type: "linear", dateBreaks: "1 year" }` throws `SpecValidationError` / `scale-type-mismatch`.

File: packages/core/src/pipeline/finalize-layout-pass.ts [hunk 15]
```diff
@@ -33,28 +33,79 @@ export function finalizePanelLayoutPass(input: {
...
+      try {
+        return parseTemporalInterval(value).key === error.value;
+      } catch {
+        return value === error.value;
+      }
+    };
+    const axis = (["x", "y"] as const).find(
+      (candidate) =>
+        matchesInterval(scalesConfig[candidate]?.dateBreaks) ||
+        matchesInterval(scalesConfig[candidate]?.dateMinorBreaks),
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/core/src/pipeline/finalize-layout-pass.ts]: ### ❌ Invalid / not actionable: already fixed on main

Why: `TemporalGuideIntervalError` carries `aesthetic` and `option` (`dateBreaks` | `dateMinorBreaks`). `finalize-layout-pass` builds `/scales/${error.aesthetic}/${error.option}` — no rematch by interval text.

File: packages/spec/src/temporal-guides.ts [hunk 16]
```diff
@@ -0,0 +1,445 @@
...
+      second: 0,
+      millisecond: 0,
+    });
+  };
+
+  let plain;
+  switch (interval.unit) {
+    case "hour": {
+      const hour = Math.floor(parts.hour / interval.step) * interval.step;
+      plain = Temporal.PlainDateTime.from({ ...parts, hour, minute: 0, second: 0, millisecond: 0 });
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/spec/src/temporal-guides.ts:335]: ### ❌ Invalid / not actionable: civil-day phase is intentional

Why: Zoned multi-hour intervals align within the civil day of `min`, then advance by the authored step. Overlapping domains that share the same civil phase already emit identical ticks in the intersection (verified for `7 hours` with mins `01:00` vs `04:00` in `America/New_York`). Changing phase when `min` crosses a step boundary is expected calendar alignment, not a silent regression.

No follow-up planned unless product wants a global epoch phase for non-divisor-of-24 hour steps.

File: packages/core/src/layout/format.ts [hunk 17]
```diff
@@ -108,6 +114,299 @@ export function formatTime(ms: number, pattern: string): string {
...
+    cachedDateTimeFormat("en-US", timezone, { timeZoneName: "short" }).formatToParts(ms),
+    "timeZoneName",
+  );
+  const longOffset = partValue(
+    cachedDateTimeFormat("en-US", timezone, { timeZoneName: "longOffset" }).formatToParts(ms),
+    "timeZoneName",
+  );
+  const offsetMatch = /^GMT([+-])(\d{2}):(\d{2})$/.exec(longOffset);
+  const offset =
+    offsetMatch === null ? "+0000" : `${offsetMatch[1]}${offsetMatch[2]}${offsetMatch[3]}`;
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/core/src/layout/format.ts:248]: Valid — fixed in #359. %z accepts GMT±HH:MM:SS historical longOffset forms (e.g. Asia/Kolkata 1880 → +0521).

File: packages/core/src/layout/temporal-guide.ts [hunk 18]
```diff
@@ -0,0 +1,501 @@
...
+function buildTicks(
+  values: readonly number[],
+  intervalValue: TemporalInterval,
+  input: TemporalAxisPlanInput,
+): AxisGuideTick[] {
+  const labels = formatTemporalTickSequence(values, {
+    ...temporalOptions(input),
+    interval: intervalValue,
+    ...(input.config.dateLabels !== undefined && { pattern: input.config.dateLabels }),
+  });
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/core/src/layout/temporal-guide.ts:211]: Valid — fixed in #359. Reversed axes format contextual labels in visual order so the leading tick keeps year/month context.

File: packages/core/src/layout/temporal-guide.ts [hunk 19]
```diff
@@ -0,0 +1,501 @@
...
+    if (previous.pos + previousHalf + MIN_TEMPORAL_LABEL_GAP_PX > current.pos - currentHalf) {
+      overlap = true;
+      break;
+    }
+  }
+  const orthogonalCap = input.orthogonalMarginCapPx ?? input.marginCapPx;
+  const marginOverflow = projected.some((tick) =>
+    input.orient === "horizontal"
+      ? tick.width / 2 > input.marginCapPx || tick.height > orthogonalCap
+      : tick.width > input.marginCapPx || tick.height / 2 > orthogonalCap,
```
[the remark this answers is not in the document]
Comment (ljodea) [reply packages/core/src/layout/temporal-guide.ts:263]: Valid — fixed in #359. Margin-overflow detection includes default tickLength+tickLabelGap (6+3) chrome, matching layout reservations.

**Comment by ljodea** [comment]
Follow-up for unrebutted post-merge findings: #359 (label/%z/margin). Pre-merge P2s and error-path P3 were already fixed on the merge head — in-thread rebuttals posted.
